### PR TITLE
fix: set the default query params based on configuration

### PR DESCRIPTION
**Pull request by AmirthavalliG**
The default query parameters to be passed on resource type specific requests and not resource instance specific request (e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters

### CODE REVIEW

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ sap.ui.define([
 	 *            (oResource.meta.profile[0]) at the requested resource
 	 * @param {string} [mParameters.defaultSubmitMode] The default SubmitMode for all bindings which are associated with this model
 	 * @param {string} [mParameters.defaultFullUrlType='uuid'] The default FullUrlType if the default submit mode is either batch or transaction
+	 * @param {array}  [mParameters.defaultQueryParameters=[]] The default query parameters to be passed to for all GET requests if there is no specific binding info. It should be of type key:value pairs in the array E.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:62]: I would recommend using `@param {object[]} [mParameters.defaultQueryParameters=[]]`

Comment (flovogt) [reply src/sap/fhir/model/r4/FHIRModel.js:62]: The default query parameters to be passed to for all GET requests

This part I understand. However the part "if there is no specific binding info". I don't understand. Do you mean. The query parameter can be override by binding specific parameters?

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 2]
```diff
@@ -26,20 +26,22 @@ sap.ui.define([
 	 * @param {sap.fhir.model.r4.FHIRModel} oModel The FHIRModel
 	 * @param {boolean} bCSRF If the FHIR service supports the csrf token
 	 * @param {string} sPrefer In which kind the FHIR service shall return the responses described here https://www.hl7.org/fhir/http.html#2.21.0.5.2
+	 * @param {array}  aDefaultQueryParams The default query parameters to be passed to for all GET requests if there is no specific resource level params. It should be of type key:value pairs in the array E.g. {'_total':'accurate'}
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:29]: see jsdoc comment in the FHIRModel.js

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 3]
```diff
@@ -420,10 +422,15 @@ sap.ui.define([
 			return "";
 		}
 
-		mParameters._format = "json";
+		// according to fhir all these kinds shall be intrepreted as json
+		if (!mParameters._format || !(mParameters._format in ["json", "application/json", "application/fhir+json"])) {
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:426]: I would recommend to add a variable in the file scope `var aSupportedFormats = ["json", "application/json", "application/fhir+json"]; 
.....
var FHIRRequestor = function(sServiceUrl, oModel, bCSRF, sPrefer , aDefaultQueryParams)`.

Comment (flovogt) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js:426]: than create a private method or a function in the file scope which tests this
`isFormatSupported(mParameters._format)`

File: tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "module": "system",
+    "target": "es6",
```
Comment (flovogt) [inline tsconfig.json:4]: We should NOT use es6 syntax. As of now, UI5 is a ES5 framework which supports also legacy browser IE11. With that ES6 can not be used out of the box. Without using transpiling (e.g. babel) the IE11 use case would not run.

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js (hunk 2 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: Right now we are using the configuration only for GET ALL .. Have updated the JSdoc..

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js (hunk 3 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: done

File: tsconfig.json (hunk 4 above)
Comment (AmirthavalliG) [reply tsconfig.json]: done

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 1 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: Right now we are using the configuration only for GET ALL .. Have updated the JSdoc..

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 5]
```diff
@@ -26,20 +27,22 @@ sap.ui.define([
...
+	 * @param {object} oDefaultQueryParams The default query parameters to be passed as part of GET ALL requests and that are not specific to any particular resource(e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
 	 * @alias sap.fhir.model.r4.lib.FHIRRequestor
 	 * @author SAP SE
 	 * @constructs {FHIRRequestor} Provides the implementation of the FHIR Requestor to send and retrieve content from a FHIR server
 	 * @protected
 	 * @since 1.0.0
 	 * @version ${version}
 	 */
-	var FHIRRequestor = function(sServiceUrl, oModel, bCSRF, sPrefer) {
+	var FHIRRequestor = function(sServiceUrl, oModel, bCSRF, sPrefer , oDefaultQueryParams) {
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:38]: remove whitespace

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 6]
```diff
@@ -516,5 +521,22 @@ sap.ui.define([
 		return mResponseHeaders;
 	};
 
+	/**
+	 * checks if the _format is part of supported types
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:525]: Start method description in upper case

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 7]
```diff
@@ -516,5 +521,22 @@ sap.ui.define([
...
+	 * according to fhir all these kinds shall be intrepreted as json
+	 * @param {string} sFormat the format in a particular request
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:527]: insert an empty line after the method description

Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:527]: Start parameter description with uppercase letter

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js (hunk 5 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: done

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js (hunk 7 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: done

Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: done

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js (hunk 6 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: done

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 8]
```diff
@@ -26,20 +27,22 @@ sap.ui.define([
 	 * @param {sap.fhir.model.r4.FHIRModel} oModel The FHIRModel
 	 * @param {boolean} bCSRF If the FHIR service supports the csrf token
 	 * @param {string} sPrefer In which kind the FHIR service shall return the responses described here https://www.hl7.org/fhir/http.html#2.21.0.5.2
+	 * @param {object} oDefaultQueryParams The default query parameters to be passed as part of GET ALL requests and that are not specific to any particular resource(e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:30]: - add space after `resource` and before `(`
- "GET ALL" its not FHIR terminology. Could we use another term? e.g. `The default query parameters to passed on resource type specific requests 
 and not resource instance specific request`

Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js]: yea makes sense.. done

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 9]
```diff
@@ -59,7 +59,7 @@ sap.ui.define([
 	 *            (oResource.meta.profile[0]) at the requested resource
 	 * @param {string} [mParameters.defaultSubmitMode] The default SubmitMode for all bindings which are associated with this model
 	 * @param {string} [mParameters.defaultFullUrlType='uuid'] The default FullUrlType if the default submit mode is either batch or transaction
-	 * @param {object} [mParameters.defaultQueryParameters={}] The default query parameters to be passed as part of GET ALL requests and that are not specific to any particular resource(e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
+	 * @param {object} [mParameters.defaultQueryParameters={}] The default query parameters to be passed on resource type specific requests and not resource instance specific request (e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:62]: `not resource instance specific requests (`

File: src/sap/fhir/model/r4/lib/FHIRRequestor.js [hunk 10]
```diff
@@ -27,7 +27,7 @@ sap.ui.define([
 	 * @param {sap.fhir.model.r4.FHIRModel} oModel The FHIRModel
 	 * @param {boolean} bCSRF If the FHIR service supports the csrf token
 	 * @param {string} sPrefer In which kind the FHIR service shall return the responses described here https://www.hl7.org/fhir/http.html#2.21.0.5.2
-	 * @param {object} oDefaultQueryParams The default query parameters to be passed as part of GET ALL requests and that are not specific to any particular resource(e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
+	 * @param {object} oDefaultQueryParams The default query parameters to be passed on resource type specific requests and not resource instance specific request (e.g /Patient?_total:accurate&_format:json). It should be of type key:value pairs. e.g. {'_total':'accurate'} -> http://hl7.org/fhir/http.html#parameters
```
Comment (flovogt) [inline src/sap/fhir/model/r4/lib/FHIRRequestor.js:30]: not resource instance specific requests (

Comment (flovogt) [reply src/sap/fhir/model/r4/lib/FHIRRequestor.js:30]: add the `s`

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 9 above)
Comment (flovogt) [reply src/sap/fhir/model/r4/FHIRModel.js:62]: ### add the `s`

**Pull request merged**